kura/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+/// Main entity in this crate is `Kura`.
+/// You should start usage of `Kura` via initialization.
+/// For example you can initialize `Kura` with full set of validations:
+/// ```
+/// use kura::Kura;
+///
+/// let kura = Kura::strict_init();
+/// ```
+
+/// High level data storage representation.
+/// Provides all necessary methods to read and write data, hides implementation details.
+struct Kura {}
+
+impl Kura {
+    /// Kura reads all transactions in all block keeping its order without any validation.
+    /// Better to use only for operations with no expectations about correctnes.
+    fn fast_init() -> Self {
+        Kura {}
+    }
+
+    /// `Kura::fast_init` with transactions and blocks validation (signatures correctness and business rules).
+    fn strict_init() -> Self {
+        Kura::fast_init()
+    }
+
+    fn store(&mut self, block: iroha::Block) -> iroha::Hash {
+        iroha::Hash {}
+    }
+
+    fn read(&mut self, hash: iroha::Hash) -> iroha::Block {
+        iroha::Block {}
+    }
+}
+
+use chashmap::CHashMap;
+/// WSV reflects the current state of the system, can be considered as a snapshot. For example, WSV
+/// holds information about an amount of assets that an account has at the moment but does not
+/// contain any info history of transaction flow.
+//TODO: can we assume that these queries https://iroha.readthedocs.io/en/latest/develop/api/queries.html
+//will be use cases to read from WSV?
+struct WorldStateView {
+    /*Structure of arrays?*/
+    accounts_assets: CHashMap<String, Vec<iroha::Asset>>,
+    accounts_inbound_transactions: CHashMap<String, Vec<iroha::Transaction>>,
+    accounts_outbound_transactions: CHashMap<String, Vec<iroha::Transaction>>,
+    accounts_all_transactions: CHashMap<String, Vec<iroha::Transaction>>,
+    assets_transactions: CHashMap<String, Vec<iroha::Transaction>>,
+}
+
+/// Representation of a consistent storage.
+struct Disk {}
+
+impl Disk {
+    fn write(&mut self, block: iroha::Block) -> Result<u64, String> {
+        Result::Err("Not implemented yet.".to_string())
+    }
+}
+
+mod iroha {
```

Comment:
This won't be the iroha module. I'll push my outline code and you can then add things to the model.